karma_chameleon/bot.py

```diff
@@ -98,20 +100,38 @@ def _check_for_self_bump(msg: dict) -> bool:
         """
         return msg["user"] in msg["text"]
 
+    def get_username_from_uid(self, uid: str) -> str:
+
+        """Fetch the username corresponding to the passed UID string."""
+        if not uid:
+            return None
+
+        try:
+            client = WebClient(token=os.environ.get("SLACK_BOT_TOKEN"))
+            result = client.users_info(user=uid)
+            return result["user"]["real_name"]
+
+        except SlackApiError as e:
+            self.logger.error("Error fetching username for {}: {}".format(uid, e))
+            return None
```

Comment:
So you are suggesting that we raise the error here instead of just dealing with it?  My only concern there is that that may cause the bot to crash.  We're already logging the failure… 

Looking around in other places in the code where we look for SlackApiError, we do not pass the exception up further.